configs/common/cores/arm/HPI.py

```diff
@@ -1683,6 +1683,11 @@ class HPI_MMU(ArmMMU):
 class HPI_BTB(SimpleBTB):
     numEntries = 128
     tagBits = 18
+    associativity = 1
+    btbReplPolicy = LRURP()
+    btbIndexingPolicy = BTBSetAssociative(
+        size="512B", entry_size=4, assoc=associativity
```

Comment:
I feel your pain here. I understand you are carefully choosing size and entry_size to account for numEntries and the instShift.

It is 100% not your fault but I believe this is a bit fragile. What happens if someone changes numEntries in the BTB?

To overcome this problem I just created:

https://github.com/gem5/gem5/pull/1557

It will allow you to do

    btbIndexingPolicy = BTBSetAssociative(
        num_entries=numEntries, set_shift=instShiftAmt, assoc=associativity)

Would you be up for waiting for my PR to be merged?

Reply:
Thanks for the PR @giactra. It definitely makes it much easier to instantiate the structures. I will rebase on top of your pull request and update this patch.

Reply:
Updated accordingly.